src/main/java/io/quarkus/gizmo/ClassSignatureBuilderImpl.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.gizmo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.quarkus.gizmo.SignatureBuilder.ClassSignatureBuilder;
+import io.quarkus.gizmo.Type.ClassType;
+import io.quarkus.gizmo.Type.ParameterizedType;
+import io.quarkus.gizmo.Type.TypeVariable;
+
+class ClassSignatureBuilderImpl implements ClassSignatureBuilder {
+    private List<TypeVariable> typeParameters = new ArrayList<>();
+    private Type superClass = ClassType.OBJECT;
+    private List<Type> superInterfaces = new ArrayList<>();
+
+    @Override
+    public String build() {
+        StringBuilder signature = new StringBuilder();
+
+        // type params
+        if (!typeParameters.isEmpty()) {
+            signature.append('<');
+            for (TypeVariable typeParameter : typeParameters) {
+                typeParameter.appendTypeParameterToSignature(signature);
+            }
+            signature.append('>');
+        }
+
+        // superclass
+        superClass.appendToSignature(signature);
+
+        // interfaces
+        if (!superInterfaces.isEmpty()) {
+            for (Type superInterface : superInterfaces) {
+                superInterface.appendToSignature(signature);
+            }
+        }
+        return signature.toString();
+    }
+
+    @Override
+    public ClassSignatureBuilder addTypeParameter(TypeVariable typeParameter) {
+        typeParameters.add(typeParameter);
+        return this;
+    }
+
+    @Override
+    public ClassSignatureBuilder setSuperClass(ClassType superClass) {
+        this.superClass = superClass;
+        return this;
+    }
+
+    @Override
+    public ClassSignatureBuilder setSuperClass(ParameterizedType superClass) {
+        if (containsWildcard(superClass)) {
+            throw new IllegalArgumentException("A super type may not specify a wilcard");
+        }
+
+        this.superClass = superClass;
+        return this;
+    }
+
+    @Override
+    public ClassSignatureBuilder addSuperInterface(ClassType interfaceType) {
```

Comment:
Does `SuperInterfaces` make sense? Shouldn't it be `addInterface` or `implementInterface`?

Reply:
FWIW `ClassCreator` uses `interfaces()` and I wonder if we should stick to the same method names so `superClass()` and `interfaces()`.

Reply:
Well, it's a [jvms language](https://docs.oracle.com/javase/specs/jvms/se11/html/jvms-4.html#jvms-ClassTypeSignature) and it's consistent with `setSuperClass()`. That said, I would not mind if we change the method name to `addInterface()`.

Reply:
I would rather be consistent with what we do with `ClassCreator` as much as we can.

Reply:
Ok, let's use `addInterface()` then...

Reply:
I know I'm getting annoying but... for the `ClassCreator` builder, we have `superClass()` and `interfaces()` and for this builder, we have `setSuperClass()` and `addInterface()`.
Given the singular `interface()` won't compile, it might be for the best but I wanted to make sure you were are of the inconsistency.

Reply:
Well, `ClassCreator` is not a "real" builder. And `set/add` IMO make the most sense for builder methods (the other option is `with` but I think that `set/add` is more clear).

Reply:
Ah, you're talking about `io.quarkus.gizmo.ClassCreator.Builder`... Hmm.